### PR TITLE
deprecate passing non-function values to cond

### PR DESCRIPTION
**Pull request by panentheos**
`Utils.cond` currently treats values differently depending on whether they are functions or non-functions. This leads to some inconsistency in how it's used, and can open the door to some subtle or surprising bugs.

This deprecates the usage of non-function values. The implementation is unchanged for now, but passing a non-function value will result in a message on the console. This also changes all existing usages to conform to the stricter API. After some additional testing, the intention is to remove the option entirely.

### CODE REVIEW

File: src/components/NewWorkspaceModal.js [hunk 1]
```diff
@@ -134,22 +134,22 @@ export default _.flow(
...
-      [loading, spinnerOverlay],
+      [loading, () => spinnerOverlay],
       [hasBillingProjects, () => h(Modal, {
         title: Utils.cond(
-          [title, title],
-          [cloneWorkspace, 'Clone a workspace'],
-          [Utils.DEFAULT, 'Create a New Workspace']
+          [title, () => title],
+          [cloneWorkspace, () => 'Clone a workspace'],
+          [true, () => 'Create a New Workspace']
```
Comment (s-rubenstein) [inline src/components/NewWorkspaceModal.js]: Why use true instead of `Utils.DEFAULT`? I personally think the default is clearer on meaning.

File: src/components/PrivateDataExplorer.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ export default _.flow(
           p([
             `Thank you for your interest in the Baseline Health Study data. Baseline data is currently only being shared
              with Baseline Health Study partner sites, Duke and Stanford. If you are a current researcher at one of those sites,
-             please reach out to your institutional contacts for information on how to obtain access. In the future, Baseline 
+             please reach out to your institutional contacts for information on how to obtain access. In the future, Baseline
```
Comment (s-rubenstein) [inline src/components/PrivateDataExplorer.js:106]: This looks like an unrelated change. Is it for linting?

File: src/components/NewWorkspaceModal.js (hunk 1 above)
Comment (s-rubenstein) [reply src/components/NewWorkspaceModal.js]: I saw below that there are also some that just have a default value after without a conditional. Maybe these should move to that?

File: src/components/NewWorkspaceModal.js [hunk 3]
```diff
@@ -134,22 +134,22 @@ export default _.flow(
...
           disabled: errors,
           tooltip: Utils.summarizeErrors(errors),
           onClick: () => this.create()
         }, Utils.cond(
-          [buttonText, buttonText],
-          [cloneWorkspace, 'Clone Workspace'],
-          [Utils.DEFAULT, 'Create Workspace']
+          [buttonText, () => buttonText],
+          [cloneWorkspace, () => 'Clone Workspace'],
+          [true, () => 'Create Workspace']
```
Comment (s-rubenstein) [inline src/components/NewWorkspaceModal.js]: See above comment.

File: src/components/NewWorkspaceModal.js (hunk 1 above)
Comment (panentheos) [reply src/components/NewWorkspaceModal.js]: `Utils.DEFAULT` is a symbol with special meaning to `Utils.switchCase`. While it technically works here, I think it's misleading, because `cond` only looks for truthiness, so there's nothing special about `DEFAULT` as opposed to `true` or any other truthy value.

As for expressing the default, I've been pondering whether it would be good to also remove the special handling of the 'hanging function', and have everything use this form instead. Still thinking on that.

File: src/components/PrivateDataExplorer.js (hunk 2 above)
Comment (panentheos) [reply src/components/PrivateDataExplorer.js:106]: Sort of. My editor fixes trailing whitespace on save. The only reason it's not an actual lint error is because it's inside of a multiline string.

File: src/components/NewWorkspaceModal.js (hunk 1 above)
Comment (s-rubenstein) [reply src/components/NewWorkspaceModal.js]: I am happy with that change, but would prefer it moved to a new PR for simplicity and traceability's sake.

**Review by s-rubenstein — APPROVED** [review]
LGTM
